### PR TITLE
Code style updates and additional test coverage for teams

### PR DESCRIPTION
**Pull request by jrassa**

### CODE REVIEW

File: src/app/core/teams/teams.service.js [hunk 1]
```diff
@@ -19,628 +19,620 @@ const
...
-		let user = await User.findById(firstAdmin).exec();
-		user = User.filteredCopy(user);
+	// implicit team members is not enabled, or the team does not have implicit members enabled
+	if (!implicitMembersEnabled || !team.implicitMembers) {
+		// Return the team role
+		return teamRole;
+	}
 
-		// Audit the creation action
-		await auditService.audit('team created', 'team', 'create', TeamMember.auditCopy(creator), Team.auditCopy(newTeam), headers);
```
Comment (brianghig) [inline src/app/core/teams/teams.service.js:254]: @jrassa is `auditService` an unused variable now?

File: src/app/core/teams/teams.service.js [hunk 2]
```diff
@@ -19,628 +19,620 @@ const
...
-		await auditService.audit('team deleted', 'team', 'delete', TeamMember.auditCopy(user), Team.auditCopy(team), headers);
+/**
+ * Updates an existing team with fresh metadata
+ *
+ * @param team The team object to update
+ * @param updatedTeam
+ * @param user The user requesting the update
+ * @returns {Promise} Returns a promise that resolves if team is successfully updated, and rejects otherwise
+ */
+const updateTeam = async (team, updatedTeam, user, headers) => {
```
Comment (brianghig) [inline src/app/core/teams/teams.service.js]: @jrassa async without await

File: src/app/core/teams/teams.service.js [hunk 3]
```diff
@@ -19,628 +19,620 @@ const
...
-	};
+/**
+ * Adds a new member to the existing team.
+ *
+ * @param user The user to add to the team
+ * @param team The team object
+ * @param role The role of the user in this team
+ * @returns {Promise} Returns a promise that resolves if the user is successfully added to the team, and rejects otherwise
+ */
+const addMemberToTeam = async (user, team, role) => {
```
Comment (brianghig) [inline src/app/core/teams/teams.service.js]: @jrassa async without await

File: src/app/core/teams/teams.service.js (hunk 1 above)
Comment (jrassa) [reply src/app/core/teams/teams.service.js:254]: fixed

File: src/app/core/teams/teams.service.js (hunk 2 above)
Comment (jrassa) [reply src/app/core/teams/teams.service.js]: fixed

File: src/app/core/teams/teams.service.js (hunk 1 above)
Comment (jrassa) [reply src/app/core/teams/teams.service.js:254]: fixed

File: src/app/core/teams/teams.service.js (hunk 3 above)
Comment (jrassa) [reply src/app/core/teams/teams.service.js]: fixed

**Comment by brianghig** [comment]
@jrassa I don't have a great way of testing the teams functionality, but this LGTM on code inspection! Ready to merge at your leisure.

**Review by brianghig — APPROVED** [review]

**Pull request merged**